scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -44,10 +48,13 @@ def process_response(self, request, response, spider):
                     # force recalculating the encoding until we make sure the
                     # responsetypes guessing is reliable
                     kwargs['encoding'] = None
+                if self.keep_encoding_header:
+                    kwargs['flags'] = response.flags
+                    kwargs['flags'].append(b'decoded')
```

Comment:
:lipstick: In a single line: ``kwargs['flags'] = response.flags + [b'decoded']``